transactions.go

```diff
@@ -78,24 +78,23 @@ func (t *Transactions) add(block BlockID, tx Transaction) {
 
 // MarkMissing marks that the node was expected to have archived a transaction with a specified id, but
 // does not have it archived and so needs to have said transaction pulled from the nodes peers.
-func (t *Transactions) MarkMissing(id TransactionID) {
+func (t *Transactions) MarkMissing(id TransactionID) bool {
 	t.Lock()
 	defer t.Unlock()
 
-	t.markMissing(id)
+	return t.markMissing(id)
 }
 
-func (t *Transactions) BatchMarkMissing(ids ...TransactionID) {
-	t.Lock()
-	defer t.Unlock()
+func (t *Transactions) markMissing(id TransactionID) bool {
+	_, exists := t.buffer[id]
 
-	for _, id := range ids {
-		t.markMissing(id)
+	if exists {
+		return false
```

Comment:
why return bool if its never used?

Reply:
It's used in ledger.go to check that the transaction was atomically marked missing.